tox_conda/plugin.py

```diff
@@ -1,332 +1,525 @@
+from tox.execute.local_sub_process import LocalSubProcessExecuteInstance, LocalSubProcessExecutor
+from tox.plugin import impl
+from tox.plugin.spec import EnvConfigSet, State, ToxEnvRegister
+from tox.tox_env.api import StdinSource, ToxEnvCreateArgs
+from tox.tox_env.errors import Fail
+from tox.tox_env.installer import Installer
+from tox.tox_env.python.api import PythonInfo, VersionInfo
+from tox.tox_env.python.pip.pip_install import Pip
+from tox.tox_env.python.pip.req_file import PythonDeps
+from tox.tox_env.python.runner import PythonRun
+
+__all__ = []
+
+
+class CondaEnvRunner(PythonRun):
+    # Can be overridden in tests
+    @staticmethod
+    def _default_execute_instance_factory(
+        request: ExecuteRequest, options: ExecuteOptions, out: SyncWrite, err: SyncWrite
+    ):
+        return LocalSubProcessExecuteInstance(request, options, out, err)
+
+    _execute_instance_factory: Callable = _default_execute_instance_factory
+
+    def __init__(self, create_args: ToxEnvCreateArgs) -> None:
+        self._installer = None
+        self._executor = None
+        self._external_executor = None
+        self._created = False
+        super().__init__(create_args)
+
+    @staticmethod
+    def id() -> str:  # noqa A003
+        return "conda"
+
+    def _get_python(self, base_python: List[str]) -> Optional[PythonInfo]:
+        exe_path = base_python[0]
+
+        output = self._run_pure(
+            [
+                exe_path,
+                "-c",
+                (
+                    "import os, platform, sys;"
+                    "print(platform.python_implementation());"
+                    "print(platform.sys.version_info);"
+                    " print(sys.version.split(os.linesep)[0]);"
+                    "print(sys.maxsize > 2**32);print(platform.system())"
+                ),
+            ],
+            "_get_python",
+        )
+        output = output.split(os.linesep)
 
-    # Check for venv.envconfig.sitepackages and venv.config.alwayscopy here
-    envdir = venv.envconfig.envdir
-    python = get_py_version(venv.envconfig, action)
+        implementation, version_info, version, is_64, platform_name = output
 
-    if venv.envconfig.conda_env is not None:
-        env_path = Path(venv.envconfig.conda_env)
+        is_64 = bool(is_64)
+        match = re.match(
+            (
+                r"sys\.version_info\(major=(\d+), minor=(\d+), micro=(\d+), releaselevel='(\w+)',"
+                r" serial=(\d+)\)"
+            ),
+            version_info,
+        )
+        version_info = VersionInfo(
+            major=int(match.group(1)),
+            minor=int(match.group(2)),
+            micro=int(match.group(3)),
+            releaselevel=match.group(4),
+            serial=int(match.group(5)),
+        )
+        extra = {"executable_path": exe_path}
+
+        return PythonInfo(implementation, version_info, version, is_64, platform_name, extra)
+
+    @property
+    def _package_tox_env_type(self) -> str:
```

Comment:
Any reason we use virtualenv for packaging and not conda?

Reply:
I did not see any urgent need to diverge from the original tox behavior here, I have not experienced any issue by using it in conda. But happy to change if there is an issue with this. What would be preferred command here?

Reply:
Duno, would make some sense to use conda environments universally rather than virtualenv in some places and conda in others. nitpick. Ultimately I'm only advisory here. @AntoineD is the maintainer of this project, if he no longer needs to do so, I'm happy to hand over the maintainership bit to you.

Reply:
This is a good question. Happy to dig deeper how conda does the packaging, but it might take slightly longer. I will invest some time already in this PR to estimate how much work would be needed. If not too much, happy to do it now, if it took longer, I might prefer merging as it is not, and resolve it later. This is just an opinion though, happy to do either way.

Reply:
> Duno, would make some sense to use conda environments universally rather than virtualenv in some places and conda in others. nitpick. Ultimately I'm only advisory here.

I think that packaging with anaconda should be opt in because:
- some people use anaconda to get the required deps in constrained IT envs, but with no actual need for anaconda packaging.
- building an anaconda package take way more time! (the way `tox` and `conda build` work have some overlap)

> @AntoineD is the maintainer of this project, if he no longer needs to do so, I'm happy to hand over the maintainership bit to you.

I've been wondering about that for quite some time: I not longer have enough spare time for tox-conda and as I said do not know tox 4 so I guess that tox-conda deserves a better maintainer, @tibortakacs you seem to be the best candidate!

Reply:
Thank you for the insight, @AntoineD. Then, if you do not have any objection, let's keep the pip builder in this PR, and further down the line we can create a new option to enable conda build packager.

@AntoineD @gaborbernat Thank you, this is a great honor. Both `tox` and `conda` are key parts of the Python, especially in R&D communities, and `tox-conda` is the crucial glue between them. I am happy to act as the maintainer of the project, and see whether I can bear this responsibility on the level that this project deserves.